tests/test_interpolation.py

```diff
@@ -234,6 +245,14 @@ def test_env_interpolation(
         assert OmegaConf.select(cfg, key) == expected
 
 
+def test_env_is_cached() -> None:
```

Comment:
This is also something we can change if we are breaking compatibility with the env resolver.
I don't see a lot of point in caching it, do you?

Reply:
No, actually I think not caching `env` is better, because it makes it possible to change a config option by modifying an env variable, which might come in handy in some situations. In an earlier implementation I had disabled the cache for `env`, but you preferred to keep it enabled. Happy to change it if you now think otherwise.